### ISSUE TITLE
account registration and password change

### ISSUE THREAD

**Issue by ac615223s5** [hugmouse/maddy-password-reset#3]
could there be an option to reset the password using the old password rather a reset link? perhaps a way to register new accounts? would rather not log in to my server every time so this would greatly appreciated :)

**Comment by hugmouse** [comment]
Resetting (or rather changing) the password with the old one is not planned, and registering new users is outside the scope of this project. I might change the README, just in case, so as not to confuse people about what functionality my project provides :thinking: 

Involving one more way of resetting/changing the password requires some additional logic + new template and for now I'm way too lazy to implement it myself. I will keep this issue open just in case if someone wants to add this functionality to this project though.

**Comment by transxtech** [comment]
Sorry, how do I use it? I installed it but cannot figure out how to use it.

**Comment by hugmouse** [comment]
Hey @transxtech, let's go step-by-step.

**Prerequisites:**

1.  `maddy-password-reset` is designed to run **alongside** `Maddy` (for example, on the same VPS). It needs access to Maddy's data (like the credentials database, access to `maddy-ctl` utility)
2.  When you run `maddy-password-reset`, it starts a web server (listening on port `1323` by default) and serves the password reset form.

**Setup:**

Let's assume you are running Ubuntu 24.04 and already have `Maddy` installed. Now, you want to set up `maddy-password-reset`. Since pre-built binaries aren't provided, you'll need to compile it from source. You'll need `git` and the `go` compiler for this.

First, update your package list and install the necessary tools:

```bash
apt install golang git
```

Next, download the `maddy-password-reset` source code:

```bash
git clone https://github.com/hugmouse/maddy-password-reset.git
cd maddy-password-reset
```

Now, build the application:

```bash
go build .
```

Or if you want to just run it, then `go run .`.

And you will see the following build log:

```bash
root@maddy-email-demo:~/maddy-password-reset# go build .
go: downloading github.com/akyoto/cache v1.0.6
go: downloading github.com/labstack/echo/v4 v4.10.0
go: downloading modernc.org/sqlite v1.20.4
go: downloading github.com/labstack/gommon v0.4.0
go: downloading golang.org/x/crypto v0.2.0
go: downloading golang.org/x/net v0.4.0
go: downloading github.com/golang-jwt/jwt v3.2.2+incompatible
go: downloading github.com/valyala/fasttemplate v1.2.2
go: downloading golang.org/x/time v0.2.0
go: downloading golang.org/x/sys v0.3.0
go: downloading modernc.org/libc v1.22.2
go: downloading github.com/mattn/go-colorable v0.1.13
go: downloading github.com/mattn/go-isatty v0.0.16
go: downloading github.com/valyala/bytebufferpool v1.0.0
go: downloading github.com/dustin/go-humanize v1.0.0
go: downloading github.com/google/uuid v1.3.0
go: downloading modernc.org/mathutil v1.5.0
go: downloading modernc.org/memory v1.4.0
go: downloading golang.org/x/text v0.5.0
go: downloading github.com/remyoudompheng/bigfft v0.0.0-20200410134404-eec4a21b6bb0
```

Then you can run `./maddy-password-reset`:

```bash
root@maddy-email-demo:~/maddy-password-reset# ./maddy-password-reset 
2025/04/03 10:56:17 [SMTP] Debug mode enabled, not checking email template
2025/04/03 10:56:17 [Sqlite] Loading Maddy's credentials database
2025/04/03 10:56:17 [Cache] Registering cache for password resets
2025/04/03 10:56:17 [Echo] Initializing echo web server
2025/04/03 10:56:17 [Echo] Registering Go templates
2025/04/03 10:56:17 [echo] Starting Echo web server
⇨ http server started on [::]:1323
```

Finally, open your web browser and go to `http://<your-server-ip>:1323/reset` (replace `<your-server-ip>` with the actual IP address or domain name of your server). You should see the password reset page:

<img width="575" alt="Image" src="https://github.com/user-attachments/assets/093d1661-efb8-419a-a40b-da5365d83829" />

And with that you can start configuring it to your heart's content: https://github.com/hugmouse/maddy-password-reset?tab=readme-ov-file#configuration

----

Again, sorry for the confusion. Probably wasn't the best idea to open-source it, as it's used for a very specific use case

**Comment by transxtech** [comment]
Great! I just missed the last step - to run it directly. Thank you so much for this great app!

**Comment by transxtech** [comment]
Sorry, please see the below. I cannot run it:
root@vps-********:~# maddy-password-reset
maddy-password-reset: command not found
root@vps-********:~# cd maddy-password-reset
root@vps-********:~/maddy-password-reset# ls
COPYING  NOTICE  README.md  go.mod  go.sum  main  main.go  templates
root@vps-********:~/maddy-password-reset# maddy-password-reset
maddy-password-reset: command not found
root@vps-********:~/maddy-password-reset# ./maddy-password-reset
-bash: ./maddy-password-reset: No such file or directory

**Comment by hugmouse** [comment]
> Sorry, please see the below. I cannot run it: root@vps-**:~# maddy-password-reset maddy-password-reset: command not found root@vps-**:~# cd maddy-password-reset root@vps-********:~/maddy-password-reset# ls COPYING NOTICE README.md go.mod go.sum main main.go templates root@vps-**:~/maddy-password-reset# maddy-password-reset maddy-password-reset: command not found root@vps-**:~/maddy-password-reset# ./maddy-password-reset -bash: ./maddy-password-reset: No such file or directory

The `./maddy-password-reset` is the executable that is produced by `go build .` command, so run the command and you will have `maddy-password-reset` in your folder. Additionally you might have to make it executable by running something like `chmod +x maddy-password-reset`.

Or alternatively if you want to just run the app - `go run .` instead of `go build .`

**Comment by transxtech** [comment]
Sorry, I remember I built it yesterday but I have to run go build to have the executable maddy-password-reset. Now it works. Thank you very much!

**Comment by transxtech** [comment]
I have a problem in sending the reset email. Although I put the right email and password for sending email address in the main.go file, I always get error - "[SMTP] Failed to send mail -  454 4.7.0 auth: invalid credentials". Please help.

**Comment by hugmouse** [comment]
> I have a problem in sending the reset email. Although I put the right email and password for sending email address in the main.go file, I always get error - "[SMTP] Failed to send mail - 454 4.7.0 auth: invalid credentials". Please help.

It seems like either there's an issue with something you've specified in `main.go`, or there might be a problem with the authentication code itself and my recent changes.

Here are some things you can check in the meantime:

- Ensure that `SMTPMailUsername`, `SMTPMailPassword`, `SMTPMailHostname`, and `MXServer` are correctly configured and have been tested in other programs (e.g., Thunderbird)
- You might not need to use the full email address in `SMTPMailUsername` (e.g., `robot@example.com`). Instead, try just specifying the username (e.g., `robot`)

I will set up a temporary server tomorrow to test the latest version of Maddy and this project.

**Comment by transxtech** [comment]
Yes, I tested. The email can send email to my other emails without any problem.

**Comment by hugmouse** [comment]
Better late than never. Feature available since 2607bbee82921ca3434677d0bd0b7c5be6599007.

Also updated the README and added a new template for that.
